File: TicTakToe/c++/version1/main.cpp
#include "mainClasses.cpp"
#include "mainFunctions.cpp"

int main()
{
    string user1, user2;
    Board gameBoard;
    int row, col, result;
    row = 5;
    col = 5;
    result = 0;

    setUsername(user1, user2);
    printTheBoard(gameBoard);

    for (int i=0; i<8; i++)
    {
        printUserPrompt(gameBoard, user1, 'x', row, col);
        cout << '\n';
        setPosition(gameBoard, 'x', row, col);
        printTheBoard(gameBoard);
        result = checkWin(gameBoard, 'x');
        if (result==1)
        {
            cout << user1 << "wins!";
            break;
        }

        printUserPrompt(gameBoard, user2, 'y', row, col);
        cout << '\n';
        setPosition(gameBoard, 'y', row, col);
        printTheBoard(gameBoard);
        result = checkWin(gameBoard, 'y');
        if (result==1)
        {
            cout << user2 << "wins!";
            break;
        }

    }
    if (result == 0)
    {
        cout << "Game tied!";
    }

}

File: TicTakToe/python/version1/main.py
from TicTacToe import Board

def main():
    user1 = input('Enter first user name ')
    if not user1:
        user1 = 'Player 1'
    user2 = input('Enter second user name ')
    if not user2:
        user2 = 'Player 2'
    board = Board()

    while not board.is_full() :
        board.print_user_prompt(user1, 'X')
        board.print_the_board()
        if board.check_win('X'):
            break

        board.print_user_prompt(user2, 'O')
        board.print_the_board()
        if board.check_win('O'):
            break
    else:
        print("game tie!")

if __name__ == '__main__':
    main()
